website/api/views.py

```
@@ -1031,3 +1034,203 @@ def post(self, request, *args, **kwargs):
+        else:
+            # Only public, active, non-expired jobs for anonymous users
+            now = timezone.now()
+            return Job.objects.filter(is_public=True, status="active").filter(
+                Q(expires_at__isnull=True) | Q(expires_at__gt=now)
+            )
+
+    def perform_create(self, serializer):
+        """Set the organization and posted_by when creating a job"""
+        org_id = self.request.data.get("organization")
+        if not org_id:
+            raise ParseError("Organization ID is required")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)